docs/DeveloperGuide.md

```diff
@@ -257,329 +257,368 @@ _{Explain here how the data archiving feature will be implemented}_
+
+**MSS**
+
+1. User opens the app
+2. TAB displays quick start instructions
+
+   Use case ends.
+   
+<br/>
+
+#### Use Case: UC8 – Get Help
 
 **MSS**
 
-1. User makes a request that <u>modifies the storage. E.g. ladd, add, edit, delete,...</u>.
-2. TAB modifies the storage.
-3. User requests to undo the last modification.
-4. TAB undoes the last command.
+1. User requests for help
+2. TAB displays a summary table of all the command usages as well as a link to an online user guide page
+
+   Use case ends.
+
+<br/>
+
+#### Use Case: UC9 - Undo an Accidental Modification
+
+**MSS**
+
+1. User requests to undo a previous modification to TAB's data within the session
+2. TAB undoes the previous modification
 
    Use case ends.
 
 **Extension**
-* 1a. User did not make any request or User requests for commands that does not modify storage.
 
-  * 1a1. TAB shows an error message.
+* 1a. There are no previous modifications
+
+  * 1a1. TAB shows an error message
   
-    User case ends.
+    Use case ends.
+
+<br/>
 
-**Use Case: Redo a Command**
+#### Use Case: UC10 - Redo an Undone Modification
 
 **MSS**
 
-1. User requests to redo a command.
-2. TAB redoes the previous undone command and modifies the storage accordingly.
+1. User requests to redo a modification to TAB's data that was previously undone within the session
+2. TAB reverts the previous undo command
 
    Use case ends.
 
 **Extension**
-* 1a. User did not request for undo command within the session.
 
-    * 1a1. TAB shows an error message for no command to redo.
+* 1a. There are no previous undo requests within the session
+
+    * 1a1. TAB shows an error message
   
       Use case ends.
     
-
+<br/>
 
 ### Non-Functional Requirements
 
-1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
-2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
-4. Should be portable, i.e. user can transfer data from one device to another easily.
-5. Should allow users to easily navigate the app interface.
-6. Data should only be stored locally to ensure user privacy.
+1. Should work on any _mainstream OS_ as long as it has Java `11` or above installed
+2. The response to any user action should become visible within 2 seconds
+3. Should be able to hold up to 50 students and 50 lessons without a noticeable sluggishness in performance for typical usage
+4. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse
+5. The source code should be open source
+6. The product is offered as a free offline service
+7. The product should not require any internet connectivity
```

Comment:
Does this mean essential features do not require internet connectivity and the "online user guide page" (line 556) is a non-essential feature?

Reply:
Gd point. I'll take out this NFR